### ISSUE TITLE
nodes will retain the settings of an area3d even when they leave it if they are reparented while within said area3d.

### ISSUE THREAD

**Issue by lolmasplay** [godotengine/godot#87693]
### Tested versions

- Reproducible in versions 4.2.1 stable, 4.1.1 stable and 4.0 stable.

### System information

Godot v4.2.1.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 Laptop GPU (NVIDIA; 30.0.15.1274) - 12th Gen Intel(R) Core(TM) i5-12500H (16 Threads)

### Issue description

The effects of an area3d will persist outside of that area even though the node is no longer in the area if the node is reparented. 

### Steps to reproduce

- create a scene with a rigidbody3d and an area3d. Make sure that both have collisionshape3Ds.
- add a script that reparents the rigidbody3d while it's in the area3d. 
- playtest to see results.

The provided MRP does this with the space bar allowing you to reload the scene. The box should stop moving after leaving the area3D but it doesn't.

### Minimal reproduction project (MRP)

[bug mrp.zip](https://github.com/godotengine/godot/files/14079275/bug.mrp.zip)

**Comment by AThousandShips** [comment]
Can you please try [4.2.2.rc1](https://godotengine.org/article/release-candidate-godot-4-1-4-and-4-2-2-rc-1/)

This sounds like a duplicate of this already fixed bug:
* https://github.com/godotengine/godot/issues/77682

**Comment by lolmasplay** [comment]
> Can you please try [4.2.2.rc1](https://godotengine.org/article/release-candidate-godot-4-1-4-and-4-2-2-rc-1/)
> 
> This sounds like a duplicate of this already fixed bug:
> 
>     * [Area3D gravity problem in Godot 4 #77682](https://github.com/godotengine/godot/issues/77682)

This is right. I apologize for reporting a pre existing/already solved bug.
